examples/r-api/README.md

```diff
@@ -0,0 +1,118 @@
+
+## What is Plumber?
+<a href="https://www.rplumber.io/index.html" target='_blank' rel='noopener'>Plumber</a> is a widely used package for writing APIs in R.  Plumber is easy to use and flexible. It provides a convenient and reliable way for R to communicate with other software and services.
+You can convert a simple R functions to a responsive API by adding few comments. Decorate your existing code with ‘#*’ . By adding these special comments as a prefix you communicate to Plumber that 'these R functions have to be transformed to API endpoints'. 
+## Objective
+In the example below we are building an API as R script (`.R` files) using Plumber. For this example we are creating a regression model. The data for same is taken from [Kaggle](https://www.kaggle.com/c/house-prices-advanced-regression-techniques/data).
+We will be accepting number of bedrooms and year build as inputs. We have defined an endpoint, "/predict", which will return the predicted price of a house.
+. 
+## Building an API
+First we will create an R script which installs plumber package from CRAN. Let's name this file as install-packages.R.
+
+```R
+install.packages('plumber')
+install.packages("dplyr")
+```
+
+Now let us create another R script room.R, this script will contain our regression model, which we will wrap with plumber. In the following code in first line we have placed oxigen2-like comments for describing what this API will do. Now we will define operation and path . Here we have used `GET` operation (read data) and path is `/predict`. This endpoint will return the predicted price of a house.
+
+This function is taking 3 arguments: `response`, `bedrooms` which represents number of bedrooms and `year` which represents year build. `bedrooms` and `year` are inputs from the client.
+
+Inside function we read houseprice data and perform linear regression to build a model. We then predict price of house for given inputs. We are returning HTTP response 400 if parameters passed are not in range of training data. If the parameters fall within valid range, API will return predicted house price.
+
+```R
+#* Predict price of a house
+#* @get /predict
+function(bedrooms,year,res){
+  data <- read.csv("https://saturn-public-data.s3.us-east-2.amazonaws.com/examples/dashboard/housePriceData.csv")
+  l = lm(SalePrice~BedroomAbvGr+YearBuilt, data)
+  bedrooms<-as.numeric(bedrooms)
+  year<-as.numeric(year)
+  if(between(bedrooms,0, 8) & between(year, 1871,2100)){
+    X <- data.frame(BedroomAbvGr=bedrooms,YearBuilt=year)
+    return(predict(l, newdata = X))
+    
+  } else {
+    res$status <- 400  
+    return(list(error = "Please enter BedroomAbvGr between 0 and 8. Enter YearBuilt between 1872 and 2100."))
+  }  
+}
```

Comment:
I moved the model fitting outside the API call so you don't have to rerun the model after each prediction. I also renamed the model to a more clear variable name.

```suggestion
data <- read.csv("https://saturn-public-data.s3.us-east-2.amazonaws.com/examples/dashboard/housePriceData.csv")
model <- lm(SalePrice~BedroomAbvGr+YearBuilt, data)

#* Predict price of a house
#* @get /predict
function(bedrooms,year,res){
  bedrooms<-as.numeric(bedrooms)
  year<-as.numeric(year)
  if(between(bedrooms,0, 8) && between(year, 1871,2100)){
    X <- data.frame(BedroomAbvGr=bedrooms, YearBuilt=year)
    return(predict(model, newdata = X))
  } else {
    res$status <- 400  
    return(list(error = "Please enter BedroomAbvGr between 0 and 8. Enter YearBuilt between 1872 and 2100."))
  }  
}
```